src/ray/raylet/worker_pool_test.cc

```diff
@@ -36,8 +36,9 @@ class WorkerPoolMock : public WorkerPool {
  public:
   explicit WorkerPoolMock(boost::asio::io_service &io_service,
                           const WorkerCommandMap &worker_commands)
-      : WorkerPool(io_service, POOL_SIZE_SOFT_LIMIT, 0, MAXIMUM_STARTUP_CONCURRENCY, 0, 0,
-                   {}, nullptr, worker_commands, []() {}),
+      : WorkerPool(io_service, NodeID::FromRandom(), "", POOL_SIZE_SOFT_LIMIT, 0,
```

Comment:
Can you add a test case please?

Reply:
Unit test that fails to kill idle workers right?

Reply:
Yes

Reply:
So, it seems like this feature wasn't unit tested in the worker pool so far. (`TryKilllingIdleWorkers` was a private method), and it tested the feature primarily in python integration tests.

If you'd still want me to add tests, I can do that, but we probably need to write tests for the feature itself first in this case.

Reply:
Why don't we do this in a separate PR in this case? I can assign time for this next sprint.

Reply:
Can you give it a shot here? Given the worker pool tests already exists, it probably isn't more than a few dozen lines needed.

Reply:
Sure. I can do that if you prefer that.

Reply:
Okay. It turns out it wasn't that simple, but I managed to do that anyway. Pushed the change.